packages/loopring_v3/contracts/impl/libexchange/ExchangeBlocks.sol

```diff
@@ -336,6 +188,7 @@ library ExchangeBlocks
             // Prepare the data for batch verification
             uint[] memory publicInputs = new uint[](batchLength);
             uint[] memory proofs = new uint[](batchLength * 8);
+
```

Comment:
Here for loop of blocks.length. To save GAS, we can limit the parameters passed from operator must be arranged in order already according to block type. So for each block type, we only need to judge whether the next block type is bigger than previous block type.

Reply:
Blocks need to be processed sequentially, so if the order in which the blocks need to be committed differ from which they can be batched more efficiently than there always needs to be some re-ordering, re-mapping or extra input validation, it just depends on where you do it or where you take the cost. We could do things slightly differently here to save some gas, but then `submitBlocks` would be much less clean than it is now, so I think it won't be worth it. 

We could also just require that all blocks that are submitted can all be batch verified together, but that makes some scenario's less efficient.